### PR TITLE
Update documentation to replace Apache Arrow DataFusion with Apache DataFusion

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/9691

## Rationale for this change

DataFusion is now a top-level project

## What changes are included in this PR?

Documentation changes only.

There are no changes to links or release process yet.

## Are these changes tested?

No

### CODE REVIEW

File: datafusion/proto/README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@
   under the License.
 -->
 
-# `datafusion-proto`: Apache Arrow DataFusion Protobuf Serialization / Deserialization
+# `datafusion-proto`: Apache DataFusion Protobuf Serialization / Deserialization
 
 This crate contains code to convert Apache Arrow [DataFusion] plans to and from
```
Comment (phillipleblanc) [inline datafusion/proto/README.md:22]:
```suggestion
This crate contains code to convert Apache [DataFusion] plans to and from
```

File: dev/release/create-tarball.sh [hunk 2]
```diff
@@ -80,11 +80,10 @@ echo ""
 echo "---------------------------------------------------------"
 cat <<MAIL
 To: dev@arrow.apache.org
```
Comment (phillipleblanc) [inline dev/release/create-tarball.sh:82]:
```suggestion
To: dev@datafusion.apache.org
```

File: dev/release/create-tarball.sh [hunk 3]
```diff
@@ -98,9 +97,9 @@ encouraged to test the release and vote with "(non-binding)".
 
 The standard verification procedure is documented at https://github.com/apache/arrow-datafusion/blob/main/dev/release/README.md#verifying-release-candidates.
```
Comment (phillipleblanc) [inline dev/release/create-tarball.sh:98]:
```suggestion
The standard verification procedure is documented at https://github.com/apache/datafusion/blob/main/dev/release/README.md#verifying-release-candidates.
```

File: dev/release/create-tarball.sh [hunk 4]
```diff
@@ -133,4 +132,4 @@ gpg --armor --output ${tarball}.asc --detach-sig ${tarball}
 echo "Uploading to apache dist/dev to ${url}"
 svn co --depth=empty https://dist.apache.org/repos/dist/dev/arrow ${SOURCE_TOP_DIR}/dev/dist
```
Comment (phillipleblanc) [inline dev/release/create-tarball.sh:133]: Should this be updated too?

**Review by phillipleblanc — APPROVED** [review]

File: dev/release/create-tarball.sh (hunk 2 above)
Comment (andygrove) [reply dev/release/create-tarball.sh:82]: This mailing list isn't active yet. This PR is just to update references to the product name.

File: dev/release/create-tarball.sh (hunk 3 above)
Comment (andygrove) [reply dev/release/create-tarball.sh:98]: The repo hasn't been renamed yet.

File: dev/release/create-tarball.sh (hunk 4 above)
Comment (andygrove) [reply dev/release/create-tarball.sh:133]: Not yet. There will be a separate PR after the repo is renamed for updating the release aspects.

**Review by comphead — APPROVED** [review]
Thanks @andygrove awesome

**Comment by comphead** [comment]
We should have separate PR on repo specific things, like paths, but this gonna be when code moved to the new repo, imho

**Comment by alamb** [comment]
> We should have separate PR on repo specific things, like paths, but this gonna be when code moved to the new repo, imho

I have filed https://github.com/apache/arrow-datafusion/issues/10134 to track this

File: dev/release/create-tarball.sh (hunk 2 above)
Comment (alamb) [reply dev/release/create-tarball.sh:82]: There is a 24 hour waiting period between requesting a new mailing list and when it is created. I think the mailing lists will be active tomorrow

**Review by alamb — APPROVED** [review]
Thank you @andygrove  (and @comphead  and @phillipleblanc for the reviews)

**Pull request merged**